src/main/java/org/opensearch/knn/training/TrainingJobRunner.java

```diff
@@ -0,0 +1,174 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.knn.training;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.index.IndexResponse;
+import org.opensearch.knn.indices.ModelDao;
+import org.opensearch.knn.indices.ModelMetadata;
+import org.opensearch.knn.indices.ModelState;
+import org.opensearch.threadpool.ThreadPool;
+
+import java.io.IOException;
+import java.util.concurrent.RejectedExecutionException;
+import java.util.concurrent.Semaphore;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import static org.opensearch.knn.common.KNNConstants.TRAIN_THREAD_POOL;
+
+/**
+ * TrainingJobRunner is a singleton class responsible for submitting TrainingJobs to the k-NN training pool executor.
+ * Capacity of queue and number of threads of the executor can be configured from executor construction (in KNNPlugin).
+ */
+public class TrainingJobRunner {
+
+    public static Logger logger = LogManager.getLogger(TrainingJobRunner.class);
+
+    private static TrainingJobRunner INSTANCE;
+    private static ModelDao modelDao;
+    private static ThreadPool threadPool;
+
+    private final Semaphore semaphore;
+    private final AtomicInteger jobCount;
+
+    /**
+     * Get singleton instance of TrainingJobRunner
+     *
+     * @return singleton instance of TrainingJobRunner
+     */
+    public static synchronized TrainingJobRunner getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new TrainingJobRunner();
+        }
+        return INSTANCE;
+    }
+
+    private TrainingJobRunner() {
+        this.jobCount = new AtomicInteger(0);
+        this.semaphore = new Semaphore(1);
+    }
+
+    /**
+     * Initializes static components.
+     *
+     * @param threadPool threadPool to use to get KNN Training Executor
+     * @param modelDao modelDao used to serialize the models
+     */
+    public static void initialize(ThreadPool threadPool, ModelDao modelDao) {
+        TrainingJobRunner.threadPool = threadPool;
```

Comment:
Why not make it instance variable? Anyway there will be one instance due to singleton instance